crates/sui-graphql-rpc/src/types/epoch.rs

```diff
@@ -12,52 +12,121 @@ use super::transaction_block::{TransactionBlock, TransactionBlockFilter};
 use super::validator_set::ValidatorSet;
 use async_graphql::connection::Connection;
 use async_graphql::*;
+use sui_indexer::models_v2::epoch::StoredEpochInfo;
+use sui_types::sui_system_state::sui_system_state_summary::SuiValidatorSummary;
 
-#[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
-#[graphql(complex)]
+// #[derive(Clone, Debug, PartialEq, Eq, SimpleObject)]
+// #[graphql(complex)]
+#[derive(Clone, Debug)]
 pub(crate) struct Epoch {
-    /// The epoch's id as a sequence number that starts at 0 and it is incremented by one at every epoch change
-    pub epoch_id: u64,
+    pub stored_epoch_info: StoredEpochInfo,
+    pub validators_summary: Vec<SuiValidatorSummary>,
+}
+
+#[Object]
+impl Epoch {
     /// The epoch's protocol version
     #[graphql(skip)]
-    pub protocol_version: u64,
+    pub fn protocol_version(&self) -> u64 {
+        self.stored_epoch_info.protocol_version as u64
+    }
+
+    /// The epoch's id as a sequence number that starts at 0 and it is incremented by one at every epoch change
+    async fn epoch_id(&self) -> u64 {
+        self.stored_epoch_info.epoch as u64
+    }
+
     /// The minimum gas price that a quorum of validators are guaranteed to sign a transaction for
-    pub reference_gas_price: Option<BigInt>,
+    async fn reference_gas_price(&self) -> Option<BigInt> {
+        Some(BigInt::from(
+            self.stored_epoch_info.reference_gas_price as u64,
+        ))
+    }
+
     /// Validator related properties, including the active validators
-    pub validator_set: Option<ValidatorSet>,
+    async fn validator_set(&self) -> Option<ValidatorSet> {
+        let active_validators = convert_to_validators(self.validators_summary.clone(), None);
```

Comment:
is convert_to_validators used elsewhere? we could move it into this file

Reply:
Let me do this in a follow-up PR. It might get used in `validator.rs` later.